lib/launcher.ts

```diff
@@ -108,197 +107,170 @@ let initFn = function(configFile: string, additionalConfig: Config) {
-            throw new Error('Must specify only 1 browser while using elementExplorer');
-          } else {
-            config.capabilities = config.multiCapabilities[0];
-          }
-          config.framework = 'explorer';
+  // 2) Set `multicapabilities` using `capabilities`,
+  // `multicapabilities`, or default
+  if (config.capabilities) {
+    if (config.multiCapabilities.length) {
+      logger.warn(
+          'You have specified both capabilities and ' +
+          'multiCapabilities. This will result in capabilities being ' +
+          'ignored');
+    } else {
+      // Use capabilities if multiCapabilities is empty.
+      config.multiCapabilities = [config.capabilities];
+    }
+  } else if (!config.multiCapabilities.length) {
+    // Default to chrome if no capabilities given
+    config.multiCapabilities = [{browserName: 'chrome'}];
+  }
 
-          let runner = new Runner(config);
-          return runner.run().then(
-              (exitCode: number) => {
-                process.exit(exitCode);
-              },
-              (err: Error) => {
-                logger.error(err);
-                process.exit(1);
-              });
-        }
-      })
-      .then(() => {
-        // 4) Run tests.
-        let scheduler = new TaskScheduler(config);
+  // 3) If we're in `elementExplorer` mode, run only that.
+  if (config.elementExplorer || config.framework === 'explorer') {
+    if (config.multiCapabilities.length != 1) {
+      throw new Error('Must specify only 1 browser while using elementExplorer');
+    } else {
+      config.capabilities = config.multiCapabilities[0];
+    }
+    config.framework = 'explorer';
+    try {
+      let runner = new Runner(config);
+      let exitCode = await runner.run();
+      process.exit(exitCode);
+    } catch (err) {
+      logger.error(err);
+      process.exit(1);
+    }
+  }
 
-        process.on('uncaughtException', (exc: (Error|string)) => {
-          let e = (exc instanceof Error) ? exc : new Error(exc);
-          if (config.ignoreUncaughtExceptions) {
-            // This can be a sign of a bug in the test framework, that it may
-            // not be handling WebDriver errors properly. However, we don't
-            // want these errors to prevent running the tests.
-            logger.warn('Ignoring uncaught error ' + exc);
-            return;
-          }
+  // 4) Run tests.
+  let scheduler = new TaskScheduler(config);
 
-          let errorCode = ErrorHandler.parseError(e);
-          if (errorCode) {
-            let protractorError = e as ProtractorError;
-            ProtractorError.log(logger, errorCode, protractorError.message, protractorError.stack);
-            process.exit(errorCode);
-          } else {
-            logger.error(e.message);
-            logger.error(e.stack);
-            process.exit(ProtractorError.CODE);
-          }
-        });
+  // process.on('uncaughtException', (exc: (Error|string)) => {
```

Comment:
Why is this commented out? Well I don't think this is ever called. The ErrorHandler is now parsing errors in ~line 242-ish.